### PR TITLE
feat: support ASR generation of multiple control statements in DoConcurrentLoop

### PR DESCRIPTION
**Pull request by adit4443ya**
Extends [PR](https://github.com/lfortran/lfortran/pull/4920) and towards [Issue](https://github.com/lfortran/lfortran/issues/4900)
Next phase of implementing Multiple Control Statements

### CODE REVIEW

File: integration_tests/openmp_36.f90 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+program openmp_36
```
Comment (Pranavchiku) [inline integration_tests/openmp_36.f90:1]: If this works with gfortran then register it in integration test without `llvm` label and register a corresponding reference test to generate ASR for LFortran

File: src/libasr/codegen/asr_to_julia.cpp [hunk 2]
```diff
@@ -1237,9 +1237,11 @@ class ASRToJuliaVisitor : public ASR::BaseVisitor<ASRToJuliaVisitor>
 
     void visit_DoConcurrentLoop(const ASR::DoConcurrentLoop_t& x)
     {
-        LCOMPILERS_ASSERT(x.n_head == 1);
-        const ASR::DoLoop_t do_loop = ASR::DoLoop_t{ x.base, nullptr, x.m_head[0], x.m_body, x.n_body, nullptr, 0 };
+        // LCOMPILERS_ASSERT(x.n_head == 1);
+        for (size_t i = 0; i < x.n_head; i++) {
+        const ASR::DoLoop_t do_loop = ASR::DoLoop_t{ x.base, nullptr, x.m_head[i], x.m_body, x.n_body, nullptr, 0 };
         visit_DoLoop(do_loop, true);
+        }
```
Comment (Pranavchiku) [inline src/libasr/codegen/asr_to_julia.cpp:1244]: Here, we need to see if `julia` has something similar to do concurrent. The current logic will create `n` number of do loops with same body where `n` is number of control statements

File: src/libasr/pass/openmp.cpp [hunk 3]
```diff
@@ -792,7 +792,7 @@ class DoConcurrentVisitor :
             DoConcurrentStatementVisitor v(al, current_scope);
             v.current_expr = nullptr;
             v.visit_DoConcurrentLoop(do_loop);
-            LCOMPILERS_ASSERT(do_loop.n_head == 1);
+            // LCOMPILERS_ASSERT(do_loop.n_head == 1);
```
Comment (Pranavchiku) [inline src/libasr/pass/openmp.cpp:795]: just remove line, don't add as a comment

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
left initial comments, the logic seems fine to me.

File: src/libasr/codegen/asr_to_julia.cpp (hunk 2 above)
Comment (certik) [reply src/libasr/codegen/asr_to_julia.cpp:1244]: I would not lose too much time with Julia right now, any workaround there is fine.

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Update reference tests and CI shall pass:

```console
rm -rf ./tests/reference/*
./run_tests.py -u
```

**Review by Pranavchiku — APPROVED** [review]

**Comment by Pranavchiku** [comment]
I am sorry I by mistake merged it. I had to do `squash and merge`. I apologise this was lack of attention while merging and assure this won't happen in future.

**Pull request merged**